mlflow/lightgbm.py

```diff
@@ -379,6 +380,22 @@ def log_feature_importance_plot(features, importance, importance_type):
         else:
             kwargs["callbacks"] = [callback]
 
+        # We set free_raw_data to false on the Dataset object
+        # so that we can access the original data later.
+        train_data = args[1] if len(args) > 1 else kwargs["train_data"]
```

Comment:
I don't think we ended up addressing the previous comment here; i.e., what happens if `len(args) <= 1` and `train_data` isn't in kwargs? In that case, it looks like MLflow will throw before LightGBM does, which creates a somewhat confusing UX.

Reply:
makes sense; for some reason i thought indexing would return None if not found

I switched it to .get instead so that it will not throw, and the code can pass to the original lightGBM train function and give the correct error message